---
title: "Proteomics"
date: "2020-01-28"
author: Jacques SERIZAY
bibliography: bibliography.bib
---

# Proteomics

IMPORTANT: This project has been written by Jacques SERIZAY, for the purpose of 
analysing public datasets. 

---  

Today you will be focusing on analyzing the proteomics data generated in 
Reinke et al., 2017: "In vivo mapping of tissue- and 
subcellular-specific proteomes in Caenorhabditis elegans.",



## Set up environment
```{r collapse = TRUE, eval = FALSE}
    PROJECT_PATH <- getwd()
    require(tidyverse)
    require(rtracklayer)
    require(org.Ce.eg.db)
    require(biomaRt)
```

---

## Load and inspect the data
The data has been fetched from [online](https://advances.sciencemag.org/highwire/filestream/195514/field_highwire_adjunct_files/1/1602426_TableS4.xlsx) 
and reformatted for the purpose of the course. Get it in ./data/1602426_TableS4.csv. 

### Questions: 
- How many proteins are studied?
- How many are in each tissue? 

```{r collapse = TRUE, eval = FALSE}
    proteomics <- read.csv(
        file.path(PROJECT_PATH, 'data/1602426_TableS4_processed.csv'), 
        na.strings = "NA"
    )
    str(proteomics)
```

---

## Adding gene information
This file only contains the protein names. Eventually, we will be 
looking at genes as well. The simplest approach is to convert everything
to unique gene IDs. 
For this, we will rely on BiomaRt.

### Questions: 
- What is the difference between Swissprot and Trembl? 
- Why is there less annotations from SwissProt?
- Why does the original dataset contain both types of IDs?
- Are there proteins coming from the same gene? 

```{r collapse = TRUE, eval = FALSE}
    # ensembl <- useDataset("celegans_gene_ensembl", mart = useMart("ensembl"))
    # ids <- getBM(
    #     attributes = c("uniprotswissprot", "uniprotsptrembl", "wormbase_gene"),
    #     mart = ensembl
    # )
    # You can also get the ids table from the data folder: 
    ids <- read.table('data/proteins_genes_IDs.txt', header = TRUE, sep = '\t')
    ids$protID <- paste0(ids$uniprotswissprot, ids$uniprotsptrembl)
    proteomics$WormBaseID <- ids$wormbase_gene[match(proteomics$UniProtKB, ids$protID)]
    proteomics <- proteomics[!is.na(proteomics$WormBaseID),]
```

---

## Separating into lists of tissue-specific proteins
Let's figure out which proteins are specifically present in each tissue.

### Questions: 
- For each set of proteins, are they more cytoplasmic or nuclear-enriched? 
- Which ones are transcription factors? (Hint: there is a list of all
transcription factors annotated in C. elegans in the data folder...)
- Can you say anything about the tissue-specific TFs? 
- What do you think about this? Are the number of tissue-specific TFs
consistent with the number of total TFs?

```{r collapse = TRUE, eval = FALSE}
    list_prots <- lapply(
        levels(proteomics$Tissue.specific), 
        function(TISSUE) {
            proteomics$UniProtKB %>%
                '['(proteomics$Tissue.specific == TISSUE & !is.na(proteomics$Tissue.specific)) %>% 
                as.character()
        }
    ) %>% setNames(levels(proteomics$Tissue.specific))
    list_genes <- lapply(
        levels(proteomics$Tissue.specific), 
        function(TISSUE) {
            proteomics$WormBaseID %>%
                '['(proteomics$Tissue.specific == TISSUE & !is.na(proteomics$Tissue.specific)) %>% 
                as.character()
        }
    ) %>% setNames(levels(proteomics$Tissue.specific))
    lengths(list_prots)
    tfs <- readLines('data/TFs.txt')
    lapply(list_prots, function(prots) {
        prots[proteomics$WormBaseID[match(prots, proteomics$UniProtKB)] %in% tfs]
    })
```

---

## Compare it to existing tissue-specific datasets from RNA-seq
Let's see if this set of proteins overlap with tissue-specific
gene annotation. For this, we can use the data in ./data/gene_annotations.gff3. 
This is a .gff3 file, a format used to add specific information
to gene annotations.

### Questions: 
- Explain the fundamental differences between the 2 sets of data?
- Are the experiments from the same developmental stage?
- Are the two datasets corresponding to each other?
- What are some good ways to represent the intersection between
these datasets?

```{r collapse = TRUE, eval = FALSE}
    genes <- rtracklayer::import('data/gene_annotations.gff3')
    names(genes) <- genes$ID
    genes$which.tissues <- factor(genes$which.tissues, levels = c(
        'Germline', 'Sperm', 'EarlyGermline', 'Neurons', 'Muscle', 'Hypod.', 'Intest.', 
        'Germline_Neurons', 'Germline_Muscle', 'Germline_Hypod.', 'Germline_Intest.', 'Neurons_Muscle', 'Neurons_Hypod.', 'Neurons_Intest.', 'Muscle_Hypod.', 'Muscle_Intest.', 'Hypod._Intest.',
        'Germline_Neurons_Muscle', 'Germline_Neurons_Hypod.', 'Germline_Neurons_Intest.', 'Germline_Muscle_Hypod.', 'Germline_Muscle_Intest.', 'Germline_Hypod._Intest.', 'Neurons_Muscle_Hypod.', 'Neurons_Muscle_Intest.', 'Neurons_Hypod._Intest.', 'Muscle_Hypod._Intest.', 
        'Germline_Neurons_Muscle_Hypod.', 'Germline_Neurons_Muscle_Intest.', 'Germline_Neurons_Hypod._Intest.', 'Germline_Muscle_Hypod._Intest.', 'Neurons_Muscle_Hypod._Intest.', 
        'Germline_Neurons_Muscle_Hypod._Intest.', 
        'Soma', 'Ubiq.', 'Ubiq.-Reg', 'Unclassified', 'Low', 'non-prot-cod'
    ))
    lapply(list_genes, function(g) {
        table(genes[g]$which.tissues)
    })
```

---  

## Save your work!!
```{r collapse = TRUE, eval = FALSE}
    save.image('results/Day2.RData')
```

--- 

## SessionInfo 
```{r echo=FALSE, collapse = TRUE, eval = FALSE}
    sessionInfo()
```
