<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 ATAC-seq | Introduction to Multiomics Data Integration and Visualisation</title>
  <meta name="description" content="Chapter 3 ATAC-seq | Introduction to Multiomics Data Integration and Visualisation" />
  <meta name="generator" content="bookdown  and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 ATAC-seq | Introduction to Multiomics Data Integration and Visualisation" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 ATAC-seq | Introduction to Multiomics Data Integration and Visualisation" />
  
  
  

<meta name="author" content="Jacques SERIZAY" />


<meta name="date" content="2020-02-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="proteomics.html">
<link rel="next" href="chip-seq.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="proteomics.html"><a href="proteomics.html"><i class="fa fa-check"></i><b>2</b> Proteomics</a><ul>
<li class="chapter" data-level="2.1" data-path="proteomics.html"><a href="proteomics.html#set-up-environment"><i class="fa fa-check"></i><b>2.1</b> Set up environment</a></li>
<li class="chapter" data-level="2.2" data-path="proteomics.html"><a href="proteomics.html#load-and-inspect-the-data"><i class="fa fa-check"></i><b>2.2</b> Load and inspect the data</a><ul>
<li class="chapter" data-level="2.2.1" data-path="proteomics.html"><a href="proteomics.html#questions"><i class="fa fa-check"></i><b>2.2.1</b> Questions:</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="proteomics.html"><a href="proteomics.html#adding-gene-information"><i class="fa fa-check"></i><b>2.3</b> Adding gene information</a><ul>
<li class="chapter" data-level="2.3.1" data-path="proteomics.html"><a href="proteomics.html#questions-1"><i class="fa fa-check"></i><b>2.3.1</b> Questions:</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="proteomics.html"><a href="proteomics.html#separating-into-lists-of-tissue-specific-proteins"><i class="fa fa-check"></i><b>2.4</b> Separating into lists of tissue-specific proteins</a><ul>
<li class="chapter" data-level="2.4.1" data-path="proteomics.html"><a href="proteomics.html#questions-2"><i class="fa fa-check"></i><b>2.4.1</b> Questions:</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="proteomics.html"><a href="proteomics.html#compare-it-to-existing-tissue-specific-datasets-from-rna-seq"><i class="fa fa-check"></i><b>2.5</b> Compare it to existing tissue-specific datasets from RNA-seq</a><ul>
<li class="chapter" data-level="2.5.1" data-path="proteomics.html"><a href="proteomics.html#questions-3"><i class="fa fa-check"></i><b>2.5.1</b> Questions:</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="proteomics.html"><a href="proteomics.html#save-the-work"><i class="fa fa-check"></i><b>2.6</b> Save the work!!</a></li>
<li class="chapter" data-level="2.7" data-path="proteomics.html"><a href="proteomics.html#sessioninfo"><i class="fa fa-check"></i><b>2.7</b> SessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="atac-seq.html"><a href="atac-seq.html"><i class="fa fa-check"></i><b>3</b> ATAC-seq</a><ul>
<li class="chapter" data-level="3.1" data-path="atac-seq.html"><a href="atac-seq.html#set-up-environment-1"><i class="fa fa-check"></i><b>3.1</b> Set up environment</a></li>
<li class="chapter" data-level="3.2" data-path="atac-seq.html"><a href="atac-seq.html#load-preveious-work"><i class="fa fa-check"></i><b>3.2</b> Load preveious work</a></li>
<li class="chapter" data-level="3.3" data-path="atac-seq.html"><a href="atac-seq.html#get-the-atac-seq-aging-data"><i class="fa fa-check"></i><b>3.3</b> Get the ATAC-seq aging data</a><ul>
<li class="chapter" data-level="3.3.1" data-path="atac-seq.html"><a href="atac-seq.html#questions-4"><i class="fa fa-check"></i><b>3.3.1</b> Questions:</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="atac-seq.html"><a href="atac-seq.html#investigate-the-genes-associated-to-varying-promoters-during-aging"><i class="fa fa-check"></i><b>3.4</b> Investigate the genes associated to varying promoters during aging</a><ul>
<li class="chapter" data-level="3.4.1" data-path="atac-seq.html"><a href="atac-seq.html#questions-5"><i class="fa fa-check"></i><b>3.4.1</b> Questions:</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="atac-seq.html"><a href="atac-seq.html#sessioninfo-1"><i class="fa fa-check"></i><b>3.5</b> SessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chip-seq.html"><a href="chip-seq.html"><i class="fa fa-check"></i><b>4</b> ChIP-seq</a><ul>
<li class="chapter" data-level="4.1" data-path="chip-seq.html"><a href="chip-seq.html#set-up-environment-2"><i class="fa fa-check"></i><b>4.1</b> Set up environment</a></li>
<li class="chapter" data-level="4.2" data-path="chip-seq.html"><a href="chip-seq.html#load-previous-work"><i class="fa fa-check"></i><b>4.2</b> Load previous work</a></li>
<li class="chapter" data-level="4.3" data-path="chip-seq.html"><a href="chip-seq.html#get-the-chip-seq-data"><i class="fa fa-check"></i><b>4.3</b> Get the ChIP-seq data</a><ul>
<li class="chapter" data-level="4.3.1" data-path="chip-seq.html"><a href="chip-seq.html#questions-6"><i class="fa fa-check"></i><b>4.3.1</b> Questions:</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chip-seq.html"><a href="chip-seq.html#investigate-the-tfs-associated-to-varying-promoters-during-aging"><i class="fa fa-check"></i><b>4.4</b> Investigate the TFs associated to varying promoters during aging</a><ul>
<li class="chapter" data-level="4.4.1" data-path="chip-seq.html"><a href="chip-seq.html#questions-7"><i class="fa fa-check"></i><b>4.4.1</b> Questions:</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="chip-seq.html"><a href="chip-seq.html#defining-key-tfs-involved-in-aging"><i class="fa fa-check"></i><b>4.5</b> Defining key TFs involved in aging?</a><ul>
<li class="chapter" data-level="4.5.1" data-path="chip-seq.html"><a href="chip-seq.html#questions-8"><i class="fa fa-check"></i><b>4.5.1</b> Questions:</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="chip-seq.html"><a href="chip-seq.html#sessioninfo-2"><i class="fa fa-check"></i><b>4.6</b> SessionInfo</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to Multiomics Data Integration and Visualisation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="atac-seq" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> ATAC-seq</h1>
<p>IMPORTANT: This project has been written by Jacques SERIZAY, for the purpose of
analysing public datasets.</p>
<hr />
<p>Today you will be focusing on investigating the ATAC-seq data generated in
Janes et al., 2018: “Chromatin accessibility dynamics across C. elegans
development and ageing”.</p>
<div id="set-up-environment-1" class="section level2">
<h2><span class="header-section-number">3.1</span> Set up environment</h2>
<pre class="sourceCode r"><code class="sourceCode r">    PROJECT_PATH &lt;-<span class="st"> </span><span class="kw">getwd</span>()
    <span class="kw">require</span>(tidyverse)
    <span class="kw">require</span>(rtracklayer)
    <span class="kw">require</span>(org.Ce.eg.db)
    <span class="kw">require</span>(biomaRt)</code></pre>
</div>
<div id="load-preveious-work" class="section level2">
<h2><span class="header-section-number">3.2</span> Load preveious work</h2>
<pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">load</span>(<span class="st">&#39;results/Day2.RData&#39;</span>)</code></pre>
<hr />
</div>
<div id="get-the-atac-seq-aging-data" class="section level2">
<h2><span class="header-section-number">3.3</span> Get the ATAC-seq aging data</h2>
<p>The data can be found <a href="https://elifesciences.org/articles/37344">online</a>
or reformatted for the purpose of the course in ./data/ATAC-seq.gff3</p>
<div id="questions-4" class="section level3">
<h3><span class="header-section-number">3.3.1</span> Questions:</h3>
<ul>
<li>Which format is the ATAC-seq.gff3 file? What does the file contain?</li>
<li>Look at the cluster information: can you visualize variations of
promoter accessibility during aging? // (boxplot)</li>
<li>To be even better: look into Seqplots software…</li>
<li>In which tissue(s) are the associated genes transcribed &amp; translated?
(Hint: relate to yesterday’s work…)</li>
</ul>
</div>
</div>
<div id="investigate-the-genes-associated-to-varying-promoters-during-aging" class="section level2">
<h2><span class="header-section-number">3.4</span> Investigate the genes associated to varying promoters during aging</h2>
<div id="questions-5" class="section level3">
<h3><span class="header-section-number">3.4.1</span> Questions:</h3>
<ul>
<li>What is the function of genes going up (down) during aging? // gProfiler</li>
<li>What are their counterpart in mammals? // BiomaRt</li>
<li>What are the diseases associated to these genes? // DOSE package??</li>
<li>Comment on the rationale of the analysis: we are looking at
genes associated to promoters de-regulated during aging…</li>
</ul>
<hr />
</div>
</div>
<div id="sessioninfo-1" class="section level2">
<h2><span class="header-section-number">3.5</span> SessionInfo</h2>
<pre><code>## R version 3.5.2 (2018-12-20)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.2 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] biomaRt_2.38.0       org.Ce.eg.db_3.7.0   AnnotationDbi_1.44.0
##  [4] Biobase_2.42.0       rtracklayer_1.42.2   GenomicRanges_1.34.0
##  [7] GenomeInfoDb_1.18.2  IRanges_2.16.0       S4Vectors_0.20.1    
## [10] BiocGenerics_0.28.0  forcats_0.4.0        stringr_1.4.0       
## [13] dplyr_0.8.0.1        purrr_0.3.2          readr_1.3.1         
## [16] tidyr_0.8.3          tibble_2.1.1         ggplot2_3.1.1       
## [19] tidyverse_1.2.1     
## 
## loaded via a namespace (and not attached):
##  [1] httr_1.4.0                  bit64_0.9-7                
##  [3] jsonlite_1.6                modelr_0.1.4               
##  [5] assertthat_0.2.1            blob_1.1.1                 
##  [7] GenomeInfoDbData_1.2.0      cellranger_1.1.0           
##  [9] Rsamtools_1.34.1            progress_1.2.0             
## [11] yaml_2.2.0                  pillar_1.3.1               
## [13] RSQLite_2.1.1               backports_1.1.3            
## [15] lattice_0.20-38             glue_1.3.1                 
## [17] digest_0.6.18               XVector_0.22.0             
## [19] rvest_0.3.2                 colorspace_1.4-1           
## [21] htmltools_0.3.6             Matrix_1.2-17              
## [23] plyr_1.8.4                  XML_3.98-1.19              
## [25] pkgconfig_2.0.2             broom_0.5.1                
## [27] haven_2.1.0                 bookdown_0.9.2             
## [29] zlibbioc_1.28.0             scales_1.0.0               
## [31] BiocParallel_1.16.6         generics_0.0.2             
## [33] withr_2.1.2                 SummarizedExperiment_1.12.0
## [35] lazyeval_0.2.2              cli_1.1.0                  
## [37] magrittr_1.5                crayon_1.3.4               
## [39] readxl_1.3.1                memoise_1.1.0              
## [41] evaluate_0.13               nlme_3.1-137               
## [43] xml2_1.2.0                  prettyunits_1.0.2          
## [45] tools_3.5.2                 hms_0.4.2                  
## [47] matrixStats_0.54.0          munsell_0.5.0              
## [49] DelayedArray_0.8.0          Biostrings_2.50.2          
## [51] compiler_3.5.2              rlang_0.4.2                
## [53] grid_3.5.2                  RCurl_1.95-4.12            
## [55] rstudioapi_0.10             bitops_1.0-6               
## [57] rmarkdown_1.12.6            gtable_0.3.0               
## [59] DBI_1.0.0                   R6_2.4.0                   
## [61] GenomicAlignments_1.18.1    lubridate_1.7.4            
## [63] knitr_1.22                  bit_1.1-14                 
## [65] stringi_1.3.1               Rcpp_1.0.1                 
## [67] tidyselect_0.2.5            xfun_0.5</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="proteomics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chip-seq.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
